cmd/repository.go

```diff
@@ -0,0 +1,686 @@
+	// only cares about the individual command-specific values (for example
+	// "name" and "url" when calling "add".  However, we want to provide the
+	// ability to interactively choose _all_ options if the user really wants
+	// to, therefore these prompts are only shown if the user is "confirming
+	// verbosely", for example `func repository add -cv`.  (of course the
+	// associated flags, environment variables etc are still respected.  Just
+	// no prompts unless verbose)
+	if !c.Verbose {
+		return c, nil
+	}
+
+	// Prompt the user for the "global" settings Repositories Path and Verbose.
+	// Not prompted for unless already in verbose mode (ex: func repo add -cv)
+	qs := []*survey.Question{
+		{
+			Name: "Repositories",
+			Prompt: &survey.Input{
+				Message: "Path to repositories:",
+				Default: c.Repositories,
+			},
+			Validate: survey.Required,
+		}, {
+			Name: "Verbose",
+			Prompt: &survey.Confirm{
+				Message: "Enable verbose logging:",
+				Default: c.Verbose,
+			},
+		},
+	}
+	err := survey.Ask(qs, &c)
+	return c, err
+}
+
+// command implementations
+// -----------------------
+
+// Run
+// (list by default or interactive with -c|--confirm)
+func runRepository(cmd *cobra.Command, args []string, clientFn repositoryClientFn) (err error) {
+	// Config
+	// Based on env, flags and (optionally) user pompting.
+	cfg, client, err := clientFn(args)
+	if err != nil {
+		return
+	}
+
+	// If in noninteractive, normal mode the help text is shown
+	if !cfg.Confirm {
+		return cmd.Help()
+	}
+
+	// Prompt for action to perform
+	question := &survey.Question{
+		Name: "Action",
+		Prompt: &survey.Select{
+			Message: "Operation to perform:",
+			Options: []string{"list", "add", "rename", "remove"},
+			Default: "list",
+		}}
+	answer := struct{ Action string }{}
+	if err = survey.Ask([]*survey.Question{question}, &answer); err != nil {
+		return
+	}
+
+	// Passthrough client constructor
+	// We already instantiated a client and a config using clientFn above
+	// (possibly prompting etc.), so the clientFn used for the subcommand
+	// delegation can be effectively a closure around those values so the
+	// user is not re-prompted:
+	c := func([]string) (repositoryClientConfig, RepositoryClient, error) { return cfg, client, nil }
+
+	// Run the command indicated
+	switch answer.Action {
+	case "list":
+		return runRepositoryList(args, c)
+	case "add":
+		return runRepositoryAdd(args, c)
+	case "rename":
+		return runRepositoryRename(args, c)
+	case "remove":
+		return runRepositoryRemove(args, c)
+	}
+	return fmt.Errorf("invalid action '%v'", answer.Action) // Unreachable
+}
+
+// List
+func runRepositoryList(args []string, clientFn repositoryClientFn) (err error) {
+	cfg, client, err := clientFn(args)
+	if err != nil {
+		return
+	}
+
+	// List all repositories given a client instantiated about config.
+	rr, err := client.Repositories().All()
+	if err != nil {
+		return
+	}
+
+	// Print repository names, or name plus url if verbose
```

Comment:
My inclination is that the remote URL should always be shown, and that `-v` could provide some additional information. I'm not sure what that would be - perhaps what languages are available? Not sure.  Just a thought... at this point usability concerns are mostly speculation, but imo it would be nice to always see the URL.